src/OrleansRuntime/Streams/PubSub/PubSubRendezvousGrain.cs

```diff
@@ -103,33 +101,57 @@ private static bool IsDeadProducer(IStreamProducerExtension producer)
 
         public async Task<ISet<PubSubSubscriptionState>> RegisterProducer(StreamId streamId, IStreamProducerExtension streamProducer)
         {
-            if (!IsActiveProducer(streamProducer))
-                throw new ArgumentException(String.Format("Trying to register non active IStreamProducerExtension: {0}", streamProducer.ToString()), "streamProducer");
-            
-            RemoveDeadProducers();
-            
-            var publisherState = new PubSubPublisherState(streamId, streamProducer);
-            State.Producers.Add(publisherState);
             counterProducersAdded.Increment();
```

Comment:
I wasn't 100% sure what these counters were tracking, attempts to add producers or successfully added producers.

For counterProducersTotal added I concluded that I should not increment it unless a producer was successfully added, hence it is incremented only in successful calls.  I treated counterProducersAdded as the number of attempts to add producers, regardless of failure or success.

I'm entirely ok with interpreting these differently, but their placement is dependent on what information we're trying to track.